.github/workflows/build.yml

```diff
@@ -260,15 +265,17 @@ jobs:
       run: signtool sign /f $Env:WINDOWS_CODESIGN_CERTIFICATE_PATH /p $Env:WINDOWS_CODESIGN_CERTIFICATE_PASSWORD *.msi
       working-directory: cmake_build
 
-    - name: "[macOS] Upload build to downloads.mixxx.org"
+    - name: "[macOS/Windows] Upload build to downloads.mixxx.org"
       # skip deploying Ubuntu builds to downloads.mixxx.org because these are deployed to the PPA
-      if: runner.os == 'macOS' && env.DOWNLOADS_HOSTGATOR_DOT_MIXXX_DOT_ORG_KEY_PASSWORD != null
-      run: tools/deploy.sh
+      if: runner.os != 'Linux' && github.event_name == 'push' && env.SSH_PASSWORD != null
+      run: bash tools/deploy.sh ${{ matrix.artifacts_path }}
       env:
-        FILE_TO_DEPLOY: ${{ matrix.artifacts_path }}
-        OS: ${{ runner.os }}
```

Comment:
Bring this back. Windows and macOS builds are now being deployed to the same directory https://downloads.mixxx.org/builds/gh-action-deployment/

Reply:
Yes, this makes sense IMO:

a) paths should be lowercase, using runner.OS is not
b) you can easily distinguish the OSes by the file extensions

We could include the OS in the filename, too, but why different directories? This makes it more complicated to find the download for a specific commit for the different OSes because they're not next to each other.

Reply:
>  b) you can easily distinguish the OSes by the file extensions

Just because you and I can does not mean users know the difference.

Reply:
I doubt that someone who doesn't know if his/her OS uses MSI or DMG files should use unstable builds from our download server instead of downloading the stable version via the link on the website.

But if you think it's better we can add the OS to the filename, e. g.:

- `...-windows.msi`
- `...-macos.dmg`

Reply:
We could add the OS to the filename, but that would make it harder to sort through a big listing of builds in one folder.

Reply:
If we add it to the end of the filename, I don't think it should have an impact on sort order.

Reply:
Sort order is not the issue. The issue is having an overwhelming amount of files to browse through in Apache's file browser on downloads.mixxx.org

Reply:
CTRL-F `<commitid>`? Or we create a folder for each version, but that seems overblown.

Reply:
Not a folder for each version; a folder for each OS as was already the case.

Reply:
That makes it even harder to test the same version on multiple OSes. That's worse than putting everything into the same folder.

Reply:
We didn't have separate directories per OS before: https://downloads.mixxx.org/mixxx-2.2.4/

Reply:
> That makes it even harder to test the same version on multiple OSes.

Not really and that's rarely done anyway.

> We didn't have separate directories per OS before

We do in 2.3 now before this PR.

Reply:
I think we could use another opinion here. @foss- what would you prefer, one directory with builds for every OS like https://downloads.mixxx.org/builds/2.2/release/ or a directory for each OS like https://downloads.mixxx.org/builds/mac_codesign/macOS/ ?

Reply:
Have a clear preference for OS based folders. Only in rare cases a singel person will test multiple OSs and even then it is easier to handle having all relevant (per OS) builds in one view than having everything mixed together.

Reply:
Using a directory for each version (rather than for each OS) would also have the benefit of being able to link people to a specific version (e. g. for testing). Right now we can only link to the huge list of builds.

Reply:
While I understand the idea, Is that really a problem? In most cases users will test the most recent build, so for general testing (sending the users to the most recent build) that would mean one additional click but better usability for the use care I mentioned earlier.

I don't have strong feelings about this. The question we are discussing here feels like a luxury problem to have. Much more important is having constant and reliable builds for all OSs.

Reply:
> Using a directory for each version (rather than for each OS) would also have the benefit of being able to link people to a specific version (e. g. for testing).

The only time I recall asking people to test a specific build on multiple different OSes is for QA before a release. In that rare case it is no big deal to post the link for each OS. Usually when linking a specific build it is to ask for testing on a specific OS.

Reply:
ping

Reply:
ping

Reply:
3rd ping. We haven't deployed Windows builds for weeks now

Reply:
Users [do indeed look through these directories](https://mixxx.discourse.group/t/how-to-install-2-3-0-beta-on-macos/20554). They should be easy to navigate. We should also cleanup the old builds.